pkg/kubelet/lifecycle/predicate.go

```diff
@@ -247,6 +248,31 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 	}
 }
 
+// generalFilter checks a group of filterings that the kubelet cares about.
+func (w *predicateAdmitHandler) generalFilter(ctx context.Context, pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) []PredicateFailureReason {
+	reasons := generalFilter(pod, nodeInfo)
+	hasNonAffinityErrors := false
+	for _, r := range reasons {
+		if r.GetReason() != nodeaffinity.ErrReasonPod {
+			hasNonAffinityErrors = true
```

Comment:
nit: just return early here?
```suggestion
			return reasons
```